### PR TITLE
bids2metadata now from aggregated metadata only

### PR DESCRIPTION
**Pull request by mih**

### CODE REVIEW

File: datalad/plugin/bids2scidata.py [hunk 1]
```diff
@@ -125,6 +132,15 @@
 }
 
 
+def getprop(obj, path, default):
+    """Helper to get a property from a metadata structure that might not be there"""
+    for p in path:
+        if not p in obj:
```
[the remark this answers is not in the document]
Comment (mih) [reply datalad/plugin/bids2scidata.py]: Yes. I could copy that statement into the docstring, but it just says the same thing as the code, only more vague and with more characters ;-)

File: datalad/plugin/bids2scidata.py [hunk 2]
```diff
@@ -150,88 +166,80 @@ def split_term_source_accession(val):
...
+        columns_addon.update(i.keys())
+    # convert to data frame, impose strict order
+    column_order = [
+        "Source Name",
+        "Characteristics[organism]",
+        "Characteristics[organism part]",
+        "Protocol REF",
+        "Sample Name"
+    ]
+    df = pd.DataFrame(
```
[the remark this answers is not in the document]
Comment (mih) [reply datalad/plugin/bids2scidata.py]: This PR did not introduce pandas as a dependency.

**Comment by mih** [comment]
I will have to leave this hanging here for some time. Feel free to close.

File: datalad/plugin/bids2scidata.py [hunk 3]
```diff
@@ -150,88 +166,80 @@ def split_term_source_accession(val):
...
     info = {
-        'Sample Name': fmeta.get("bids:participant_id", fname_components[0]),
+        'Sample Name': getprop(bidsmeta, ['participant', 'id'], None),
         # assay name is the entire filename except for the modality suffix
         # so that, e.g. simultaneous recordings match wrt to the assay name
         # across assay tables
-        'Assay Name': '_'.join(fname_components[:-1]),
-        'Raw Data File': fpath,
-        'Parameter Value[modality]': fname_components[-1]
+        'Assay Name': psplit(fmeta['path'])[-1].split('.')[0][:-(len(modality) + 1)],
```
[the remark this answers is not in the document]
Comment (mih) [reply datalad/plugin/bids2scidata.py]: Sadly, this statement is true for 50% of the lines in this file. There is a dedicated comment just above this line that already explains what this does. I see no need for a dedicated helper, this is the one place where this is used. Moreover, the code line is basically just moved and reduced to the stuff that is left to do after no longer working with file input.

Once we know that this command does what it is supposed to do, I will think about how to make it nice. But for now the PR just wants to make it better, not perfect.

File: datalad/plugin/bids2scidata.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mih) [reply datalad/plugin/bids2scidata.py]: Anyone who wants to work on removing the pandas dependency in the future should be aware of two things. ISATAB requires a particular substructure of its tables that involved a certain set of columns (with identical names occurring multiple times in the same document). ATM these columns are injected late in the process, and documents are merged. Any alternative implementation will either need to provide such features, or require a substantial rewrite of this command.

[the remark this answers is not in the document]
Comment (mih) [reply datalad/plugin/bids2scidata.py]: Exported to #2312

File: datalad/plugin/bids2scidata.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mih) [reply datalad/plugin/bids2scidata.py]: `not in` is done now.

**Comment by mih** [comment]
If the tests do not say otherwise, I will merge this change ASAP. This command is not done, and probably always will be a moving target. However, there is a chance to get our hand on a hand-validated ISATAB dataset that was created by the code in this PR. Once we have that, we can use it as a reference and perform more aggressive RFs.

File: datalad/plugin/bids2scidata.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (yarikoptic) [reply datalad/plugin/bids2scidata.py]: cool, thanks

File: datalad/plugin/bids2scidata.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (yarikoptic) [reply datalad/plugin/bids2scidata.py]: > Once we know that this command does what it is supposed to do, I will think about how to make it nice. 

well, ... I am just growing more pessimist every day I guess, especially about "I will get back to it and make it nice" which rarely happens and we grow on codebase harder to read/maintain. but ok

**Comment by yarikoptic** [comment]
OSX  buildbot failed with weird
```
======================================================================
FAIL: datalad.tests.test_installed.test_run_datalad_help
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/datalad/buildslave/datalad-pr-dl-osx-64/build/venv-ci/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/datalad/buildslave/datalad-pr-dl-osx-64/build/datalad/tests/utils.py", line 1290, in newfunc
    func(*args, **kwargs)
  File "/Users/datalad/buildslave/datalad-pr-dl-osx-64/build/datalad/tests/utils.py", line 1102, in newfunc
    reraise(*exc_info)
  File "/Users/datalad/buildslave/datalad-pr-dl-osx-64/build/datalad/tests/utils.py", line 1078, in newfunc
    func(*args, **kwargs)
  File "/Users/datalad/buildslave/datalad-pr-dl-osx-64/build/datalad/tests/test_installed.py", line 35, in test_run_datalad_help
    eq_(err, "")
AssertionError: 'sh: sysctl: command not found\n' != ''
```
I've not seen before... restarting, if passes - merge

**Comment by mih** [comment]
The OSX failure is not new, e.g. https://github.com/datalad/datalad/pull/2296 has the same. I think I saw it even earlier than that.

**Comment by yarikoptic** [comment]
Well, merge then.. can't do in the phone. I will need to nail it down

**Pull request merged**